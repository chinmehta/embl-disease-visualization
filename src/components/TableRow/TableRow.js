import React, { useRef, useState } from "react";
import TableCellAccordionBtn from "../TableCellAccordionBtn/TableCellAccordionBtn";
import TableCellLink from "../TableCellLink/TableCellLink";
import TableCellText from "../TableCellText/TableCellText";

function TableRow(props) {
  const BASE_API = "https://platform.opentargets.org/target/";
  const accordionRef = useRef(null);
  let [accordionText, setaccordionText] = useState("+");

  const toggleAccordion = () => {
    accordionRef.current.classList.toggle("hidden");
    accordionRef.current.classList.contains("hidden")
      ? setaccordionText("+")
      : setaccordionText("-");
  };

  return (
    <React.Fragment>
      <tr>
        <TableCellAccordionBtn
          text={accordionText}
          clickHandler={toggleAccordion}
        />
        <TableCellLink
          text={props.data.target.approvedSymbol}
          link={BASE_API + props.data.target.approvedName}
        />
        <TableCellText text={props.data.target.approvedName} />
        <TableCellText text={props.data.score.toFixed(3)} />
      </tr>
      <tr className="hidden h-96" ref={accordionRef}>
        <td colSpan="4"></td>
      </tr>
    </React.Fragment>
  );
}

export default TableRow;
