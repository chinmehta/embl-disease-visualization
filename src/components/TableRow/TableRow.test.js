import React from "react";
import renderer from "react-test-renderer";
import TableRow from "./TableRow";

it("renders when no data  is passed as prop", () => {
  const element = renderer.create(<TableRow />).toJSON();

  expect(element).toMatchSnapshot();
});

it("renders when data  is passed as prop", () => {
  const singleRow = {
    approvedSymbol: "EGFR",
    approvedName: "epidermal growth factor receptor",
    score: 0.890384,
    label: ["label1", "label2"],
    chartDataArray: [0.4, 0.9],
  };
  const element = renderer.create(<TableRow data={singleRow} />).toJSON();

  expect(element).toMatchSnapshot();
});
