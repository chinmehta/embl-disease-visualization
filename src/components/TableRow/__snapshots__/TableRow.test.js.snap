// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when data  is passed as prop 1`] = `
Array [
  <tr>
    <td
      className="flex items-stretch border border-gray-400 border-collapse p-0"
    >
      <button
        className="flex text-xl px-3 py-3 h-100 w-100 bg-embl-primary-blue text-white w-full h-full justify-center "
        onClick={[Function]}
      >
        +
      </button>
    </td>
    <td
      className="text-embl-primary-blue px-2 border-collapse border border-gray-400 leading-3"
    >
      <a
        href="https://platform.opentargets.org/target/epidermal growth factor receptor"
        rel="noopener noreferrer"
        target="_blank"
      >
        EGFR
      </a>
    </td>
    <td
      className="px-2 border-collapse border border-gray-400"
    >
      epidermal growth factor receptor
    </td>
    <td
      className="px-2 border-collapse border border-gray-400"
    >
      0.890
    </td>
  </tr>,
  <tr
    className="hidden"
  >
    <td
      className="border border-gray-400"
      colSpan="4"
    >
      <div
        className="tab-container my-2 mx-5"
      >
        <div
          className="tab-list flex border-gray-400 border-b-2"
        >
          <span
            className=" px-4 cursor-pointer bg-embl-primary-blue text-white"
            onClick={[Function]}
          >
            Bar chart
          </span>
          <span
            className=" px-4 cursor-pointer bg-white text-embl-primary-blue"
            onClick={[Function]}
          >
            Radar chart
          </span>
        </div>
        <div
          className="flex h-80 justify-center items-center"
        >
          <canvas
            className="max-h-full max-w-full"
            height={150}
            role="img"
            width={300}
          />
        </div>
      </div>
    </td>
  </tr>,
]
`;

exports[`renders when no data  is passed as prop 1`] = `
Array [
  <tr>
    <td
      className="flex items-stretch border border-gray-400 border-collapse p-0"
    >
      <button
        className="flex text-xl px-3 py-3 h-100 w-100 bg-embl-primary-blue text-white w-full h-full justify-center "
        onClick={[Function]}
      >
        +
      </button>
    </td>
    <td
      className="text-embl-primary-blue px-2 border-collapse border border-gray-400 leading-3"
    >
      <a
        href="https://platform.opentargets.org/target/undefined"
        rel="noopener noreferrer"
        target="_blank"
      />
    </td>
    <td
      className="px-2 border-collapse border border-gray-400"
    />
    <td
      className="px-2 border-collapse border border-gray-400"
    />
  </tr>,
  <tr
    className="hidden"
  >
    <td
      className="border border-gray-400"
      colSpan="4"
    >
      <div
        className="tab-container my-2 mx-5"
      >
        <div
          className="tab-list flex border-gray-400 border-b-2"
        >
          <span
            className=" px-4 cursor-pointer bg-embl-primary-blue text-white"
            onClick={[Function]}
          >
            Bar chart
          </span>
          <span
            className=" px-4 cursor-pointer bg-white text-embl-primary-blue"
            onClick={[Function]}
          >
            Radar chart
          </span>
        </div>
        <div
          className="flex h-80 justify-center items-center"
        >
          <canvas
            className="max-h-full max-w-full"
            height={150}
            role="img"
            width={300}
          />
        </div>
      </div>
    </td>
  </tr>,
]
`;
